File: app/src/main/java/com/synapse/social/studioasinc/domain/model/ReactionType.kt
package com.synapse.social.studioasinc.domain.model

import kotlinx.serialization.Serializable



@Serializable
enum class ReactionType(
    val displayName: String,
    val emoji: String
) {
    LIKE("Like", "❤️"),
    LOVE("Love", "😍"),
    HAHA("Haha", "😂"),
    WOW("Wow", "😮"),
    SAD("Sad", "😢"),
    ANGRY("Angry", "😡");

    companion object {


        fun fromString(value: String?): ReactionType {
            return values().find { it.name.equals(value, ignoreCase = true) } ?: LIKE
        }



        fun getAllReactions(): List<ReactionType> {
            return values().toList()
        }
    }
}

File: iosApp/iosApp/Models/ReactionType.swift
import Foundation

enum ReactionType: String, Codable, CaseIterable {
    case like = "LIKE"
    case love = "LOVE"
    case haha = "HAHA"
    case wow = "WOW"
    case sad = "SAD"
    case angry = "ANGRY"
    case unreact = "UNREACT"

    var emoji: String {
        switch self {
        case .like: return "👍"
        case .love: return "❤️"
        case .haha: return "😂"
        case .wow: return "😮"
        case .sad: return "😢"
        case .angry: return "😡"
        case .unreact: return ""
        }
    }
}
